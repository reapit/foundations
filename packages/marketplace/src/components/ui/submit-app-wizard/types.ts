import React from 'react'

export type WizardStep =
  | 'BEFORE_YOU_START'
  | 'INPUT_APP_NAME'
  | 'CREATE_NEW_APP'
  | 'INPUT_AUTHENTICATION_URIS'
  | 'GRANT_PERMISSION'
  | 'SUBMIT_APP_SUCCESS'

export type SetWizardStep = React.Dispatch<React.SetStateAction<WizardStep>>

export type WizardStepComponent = React.FC<{
  setWizardStep: React.Dispatch<React.SetStateAction<WizardStep>>
  afterClose: (() => void) | undefined
}>
