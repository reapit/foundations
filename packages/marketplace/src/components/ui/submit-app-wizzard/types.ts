import React from 'react'

export type WizzardStep =
  | 'BEFORE_YOU_START'
  | 'INPUT_APP_NAME'
  | 'CREATE_NEW_APP'
  | 'INPUT_AUTHENTICATION_URIS'
  | 'GRANT_PERMISSION'
  | 'SUBMIT_APP_SUCCESS'

export type SetWizzardStep = React.Dispatch<React.SetStateAction<WizzardStep>>

export type WizzardStepComponent = React.FC<{
  setWizzardStep: React.Dispatch<React.SetStateAction<WizzardStep>>
  afterClose: (() => void) | undefined
  handleUpdateFormState: any
}>
