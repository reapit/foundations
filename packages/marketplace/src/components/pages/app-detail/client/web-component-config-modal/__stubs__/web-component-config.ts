import { ReduxState } from '@/types/core'

export const webComponentStub = {
  client: {
    webComponent: { loading: false, updating: false, data: null, isShowModal: true, negotiators: {} },
  },
} as ReduxState
