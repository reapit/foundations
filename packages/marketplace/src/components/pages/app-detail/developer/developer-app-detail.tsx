import * as React from 'react'
import { useHistory } from 'react-router'
import { selectIntegrationTypes } from '@/selector/integration-types'
import { DesktopIntegrationTypeModel } from '@/actions/app-integration-types'
import { selectInstallationAppData } from '@/selector/installations'
import { PagedResultInstallationModel_ } from '@reapit/foundations-ts-definitions'
import { DeveloperAside } from './developer-aside'
import { useSelector } from 'react-redux'
import { History } from 'history'
import { selectAppDetailState, selectAppDetailData, selectAppDetailLoading } from '@/selector/developer-app-detail'
import { selectInstallAppLoading } from '@/selector/installations'
import { Loader, FlexContainerBasic, Grid, GridItem, FlexContainerResponsive } from '@reapit/elements'
import AppHeader from '@/components/pages/app-detail/common/ui-app-header'
import styles from '@/styles/blocks/standalone-app-detail.scss?mod'
import routes from '@/constants/routes'
import { getDesktopIntegrationTypes } from '@/utils/get-desktop-integration-types'
import useReactResponsive from '@/components/hooks/use-react-responsive'
import { BackToAppsSection } from '../common/ui-sections'
import AppContent from './developer-app-content'

export type DeveloperAppDetailProps = {}

export const handleOnDeleteAppSuccess = (history: History) => () => {
  history.replace(routes.DEVELOPER_MY_APPS)
}

export const closeInstallationsModal = (setIsInstallationsModalOpen: (isVisible: boolean) => void) => () => {
  setIsInstallationsModalOpen(false)
}

export const closeAppRevisionComparisonModal = (
  setIsAppRevisionComparisonModalOpen: (isVisible: boolean) => void,
) => () => {
  setIsAppRevisionComparisonModalOpen(false)
}

export const closeDeleteAppModal = (setIsDeleteModalOpen: (isVisible: boolean) => void) => () => {
  setIsDeleteModalOpen(false)
}

export const onBackToAppsButtonClick = (history: History) => () => {
  history.push(routes.DEVELOPER_MY_APPS)
}

const DeveloperAppDetail: React.FC<DeveloperAppDetailProps> = () => {
  const history = useHistory()
  const { isMobile } = useReactResponsive()

  const appDetailState = useSelector(selectAppDetailState)
  const appDetailData = useSelector(selectAppDetailData)
  const isLoadingAppDetail = useSelector(selectAppDetailLoading)
  const isLoadingInstallations = useSelector(selectInstallAppLoading)
  const desktopIntegrationTypes = useSelector(selectIntegrationTypes) as DesktopIntegrationTypeModel[]
  const installationsData = useSelector(selectInstallationAppData) as PagedResultInstallationModel_
  const unfetch = !appDetailState?.data || !installationsData?.data
  const userDesktopIntegrationTypes = getDesktopIntegrationTypes(
    appDetailData.desktopIntegrationTypeIds || [],
    desktopIntegrationTypes,
  )

  if (isLoadingAppDetail || isLoadingInstallations || unfetch) {
    return <Loader />
  }

  return (
    <FlexContainerResponsive hasPadding className={styles.flexContainer}>
      <Grid className={styles.container}>
        <GridItem className="is-one-quarter">
          <DeveloperAside desktopIntegrationTypes={userDesktopIntegrationTypes} appDetailState={appDetailState} />
        </GridItem>
        <GridItem className="is-three-quarters">
          <FlexContainerBasic flexColumn hasPadding hasBackground isFullHeight>
            <AppHeader appDetailData={appDetailData} />
            <AppContent appDetailState={appDetailState} />
            {!isMobile && <BackToAppsSection onClick={onBackToAppsButtonClick(history)} />}
          </FlexContainerBasic>
        </GridItem>
      </Grid>
    </FlexContainerResponsive>
  )
}

export default DeveloperAppDetail
