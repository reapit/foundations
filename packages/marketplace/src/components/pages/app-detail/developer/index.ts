import DeveloperAppDetail from './developer-app-detail'

export default DeveloperAppDetail
