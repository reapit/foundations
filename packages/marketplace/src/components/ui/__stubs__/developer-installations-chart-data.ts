import { InstallationModelWithAppName } from '@/components/ui/analytics/detailed/detailed-tab'

export const installedAppsStub: Array<InstallationModelWithAppName> = [
  {
    id: 'b3c2f644-3241-4298-b320-b0398ff492f1',
    appId: '062a376c-f5a3-46a0-a64b-e4bc6e5af2c1',
    created: '2020-01-31T12:57:12',
    client: 'DXX',
    status: 'Active',
    appName: 'App 1',
    links: [
      {
        rel: 'self',
        href: 'http://dev.platformmarketplace.reapit.net/installations/b3c2f644-3241-4298-b320-b0398ff492f9',
        action: 'GET',
      },
      {
        rel: 'app',
        href: 'http://dev.platformmarketplace.reapit.net/apps/062a376c-f5a3-46a0-a64b-e4bc6e5af2c1',
        action: 'GET',
      },
    ],
  },
  {
    id: 'b3c2f644-3241-4298-b320-b0398ff492f2',
    appId: '062a376c-f5a3-46a0-a64b-e4bc6e5af2c2',
    created: '2020-01-31T12:57:12',
    client: 'DXX',
    appName: 'App 2',
    status: 'Active',
    links: [
      {
        rel: 'self',
        href: 'http://dev.platformmarketplace.reapit.net/installations/b3c2f644-3241-4298-b320-b0398ff492f9',
        action: 'GET',
      },
      {
        rel: 'app',
        href: 'http://dev.platformmarketplace.reapit.net/apps/062a376c-f5a3-46a0-a64b-e4bc6e5af2c1',
        action: 'GET',
      },
    ],
  },
  {
    id: 'b3c2f644-3241-4298-b320-b0398ff492f3',
    appId: '062a376c-f5a3-46a0-a64b-e4bc6e5af2c3',
    created: '2020-02-03T01:49:57',
    client: 'DXX',
    appName: 'App 3',
    status: 'Active',
    links: [
      {
        rel: 'self',
        href: 'http://dev.platformmarketplace.reapit.net/installations/b3c2f644-3241-4298-b320-b0398ff492f9',
        action: 'GET',
      },
      {
        rel: 'app',
        href: 'http://dev.platformmarketplace.reapit.net/apps/062a376c-f5a3-46a0-a64b-e4bc6e5af2c1',
        action: 'GET',
      },
    ],
  },
]
