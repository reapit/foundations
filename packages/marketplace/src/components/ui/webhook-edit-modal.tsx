import React, { useEffect } from 'react'
import { connect } from 'react-redux'
import { Dispatch } from 'redux'
import styles from '@/styles/blocks/developer-app-modal.scss?mod'
import {
  Modal,
  Button,
  DropdownSelect,
  Input,
  Formik,
  Form,
  Content,
  Checkbox,
  SelectOption,
  ModalFooter,
  ModalBody,
  ModalHeader,
  Loader,
  Level,
  LevelLeft,
  LevelRight,
} from '@reapit/elements'
import { ReduxState } from '@/types/core'
import {
  requestWebhookSubcriptionData,
  CreateWebhookParams,
  EditWebhookParams,
  createWebhook,
  editWebhook,
  requestWebhookData,
  webhookDataClear,
  deleteWebhook,
  DeleteWebhookParams,
} from '@/actions/webhook-edit-modal'
import { WebhookModal, CustomerItem, TopicItem } from '@/reducers/webhook-edit-modal'
import { selectTopics, selectWebhookData, selectLoading, selectCustomers } from '@/selector/webhook-edit'
import { isValidHttpsUrl } from '@/utils/validate'
import linkStyles from '@/styles/elements/link.scss?mod'

const CREATE_MODAL = {
  title: 'Add New Webhook',
  submit: 'Create',
}

const EDIT_MODAL = {
  title: 'Edit',
  submit: 'Update',
}

export type WebhookEditProps = {
  isUpdate?: boolean
  appId: string
  webhookId?: string
  visible: boolean
  closeModal?: () => void
  afterClose?: () => void
}
export interface WebhookModalInnerMappedAction {
  requestWebhookSubcriptionData: (appId: string) => void
  requestWebhookData: (webhookId: string) => void
  createWebhook: (data: CreateWebhookParams) => void
  editWebhook: (data: EditWebhookParams) => void
  webhookDataClear: () => void
  deleteWebhook: (data: DeleteWebhookParams) => void
}

export const WebhookEditModal: React.FunctionComponent<WebhookEditProps> = ({
  isUpdate = false,
  appId,
  visible,
  webhookId,
  closeModal,
  afterClose,
}) => {
  if (!visible) return null
  return (
    <Modal visible={visible} renderChildren afterClose={afterClose}>
      <WebhookModalInnerWithConnect isUpdate={isUpdate} closeModal={closeModal} appId={appId} webhookId={webhookId} />
    </Modal>
  )
}

export const generateTopicOptions = (topics: TopicItem[]) => {
  return topics.map(
    topic =>
      ({
        value: topic.id,
        label: topic.name,
        description: topic.description,
      } as SelectOption),
  )
}

export const generateCustomerOptions = (customers: CustomerItem[]) => {
  const customerOptions: SelectOption[] = [
    {
      value: 'SBOX',
      label: 'SBOX',
      description: 'SBOX',
    } as SelectOption,
  ]
  customers.forEach((customer: CustomerItem) => {
    if (customer.status === 'Active') {
      customerOptions.push({
        value: customer.client,
        label: customer.client,
        description: customer.client,
      } as SelectOption)
    }
  })
  return customerOptions
}

export const onCreate = (createWebhook: (data: CreateWebhookParams) => void, appId: string) => (
  values: FormValuesType,
) => {
  const params: CreateWebhookParams = {
    applicationId: appId,
    url: values.url,
    description: '',
    topicIds: values.topicIds,
    customerIds: values.customerIds,
    active: values.active,
  }
  createWebhook(params)
}

export const onEdit = (editWebhook: (data: EditWebhookParams) => void, webhookId: string, appId: string) => (
  values: FormValuesType,
) => {
  const params: EditWebhookParams = {
    applicationId: appId,
    webhookId,
    url: values.url,
    description: '',
    topicIds: values.topicIds,
    customerIds: values.customerIds,
    active: values.active,
  }
  editWebhook(params)
}

export const validateURL = (value: string): string | null => {
  return isValidHttpsUrl(value) ? null : 'The value must be a valid and secure URI'
}

export type WebhookModalInnerProps = WebhookModalInnerMappedAction &
  StateProps & {
    isUpdate?: boolean
    webhookId?: string
    appId: string
    closeModal?: () => void
  }

export type FormValuesType = {
  url: string
  topicIds: string[]
  customerIds: string[]
  active: boolean
}

export const WebhookModalInner: React.FunctionComponent<WebhookModalInnerProps> = ({
  isUpdate = false,
  closeModal,
  topics,
  customers,
  loading,
  webhookData,
  webhookId = '',
  appId = '',
  ...props
}) => {
  const modalConfig = isUpdate ? EDIT_MODAL : CREATE_MODAL

  const initFormValues: FormValuesType = {
    url: webhookData?.url,
    topicIds: webhookData?.topicIds,
    customerIds: webhookData?.customerIds,
    active: webhookData?.active,
  }

  const topicOptions: SelectOption[] = generateTopicOptions(topics)
  const customerOptions: SelectOption[] = generateCustomerOptions(customers)

  useEffect(() => {
    if (isUpdate) {
      props.requestWebhookData(webhookId)
    } else {
      props.requestWebhookSubcriptionData(appId)
    }
    return props.webhookDataClear
  }, [])

  const onSubmit = isUpdate ? onEdit(props.editWebhook, webhookId, appId) : onCreate(props.createWebhook, appId)
  const onDelete = () => props.deleteWebhook({ webhookId, applicationId: appId })
  if (loading) return <Loader />
  return (
    <Formik initialValues={initFormValues} onSubmit={onSubmit}>
      {({ handleSubmit }) => {
        return (
          <>
            <ModalHeader title={modalConfig.title} />
            <ModalBody
              body={
                <Form>
                  <Content>
                    <p>
                      Webhooks are configured here to allow your application to receive real-time notifications about
                      the topics you choose to subscribe it to. A single webhook subscription can receive notifications
                      for multiple topics so long as your application has been granted the required permissions.
                    </p>
                    <p>
                      Webhooks subscriptions can be set up for any customer who has installed your application.
                      Additionally, you can choose ‘SBOX’ to listen for sandbox environment notifications.
                    </p>
                    <p>
                      For more information about Webhooks, please see our{' '}
                      <a
                        className={linkStyles.link}
                        href="https://foundations-documentation.reapit.cloud/api/api-documentation#webhooks"
                        target="_blank"
                        rel="noopener noreferrer"
                      >
                        webhooks documentation
                      </a>
                    </p>
                  </Content>
                  <Input
                    id="url"
                    type="text"
                    placeholder="Enter a secure URL (https://)"
                    name="url"
                    labelText="Webhook URL"
                    required
                    validate={validateURL}
                  />
                  <DropdownSelect
                    id="topicIds"
                    placeholder="Please select"
                    name="topicIds"
                    labelText="Subscription Topics"
                    options={topicOptions}
                    dropdownStyle={{ zIndex: 41 }}
                    required
                  />
                  <DropdownSelect
                    id="customerIds"
                    placeholder="All Customers who have installed your application (default)"
                    name="customerIds"
                    labelText="Subscription Customers"
                    options={customerOptions}
                    dropdownStyle={{ zIndex: 41 }}
                  />
                  <Checkbox id="active" name="active" labelText="Active" />
                </Form>
              }
            />
            <ModalFooter
              footerItems={
                <Level className={styles.footer}>
                  <LevelLeft>
                    {isUpdate && (
                      <Button className="mr-2" variant="secondary" type="button" onClick={onDelete}>
                        Delete
                      </Button>
                    )}
                  </LevelLeft>
                  <LevelRight className="mt-0">
                    <Button className="mr-2" variant="secondary" type="button" onClick={closeModal}>
                      Cancel
                    </Button>
                    <Button variant="primary" type="submit" onClick={handleSubmit}>
                      {modalConfig.submit}
                    </Button>
                  </LevelRight>
                </Level>
              }
            />
          </>
        )
      }}
    </Formik>
  )
}

export type StateProps = {
  topics: TopicItem[]
  customers: CustomerItem[]
  loading: boolean
  webhookData: WebhookModal
}

export const mapStateToProps = (state: ReduxState): StateProps => ({
  topics: selectTopics(state),
  customers: selectCustomers(state),
  loading: selectLoading(state),
  webhookData: selectWebhookData(state),
})

export const mapDispatchToProps = (dispatch: Dispatch): WebhookModalInnerMappedAction => {
  return {
    requestWebhookSubcriptionData: (appId: string) => dispatch(requestWebhookSubcriptionData(appId)),
    createWebhook: (data: CreateWebhookParams) => dispatch(createWebhook(data)),
    editWebhook: (data: EditWebhookParams) => dispatch(editWebhook(data)),
    deleteWebhook: (data: DeleteWebhookParams) => dispatch(deleteWebhook(data)),
    requestWebhookData: (webhookId: string) => dispatch(requestWebhookData(webhookId)),
    webhookDataClear: () => dispatch(webhookDataClear()),
  }
}

export const WebhookModalInnerWithConnect = connect(mapStateToProps, mapDispatchToProps)(WebhookModalInner)
export default React.memo(WebhookEditModal)
