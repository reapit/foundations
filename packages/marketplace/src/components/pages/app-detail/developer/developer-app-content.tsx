import * as React from 'react'
import { Dispatch } from 'redux'
import { GET_ALL_PAGE_SIZE } from '@/constants/paginator'
import { appInstallationsRequestData } from '@/actions/app-installations'
import { selectDeveloperId } from '@/selector'
import { FlexContainerBasic } from '@reapit/elements'
import ConfirmUninstall from '@/components/ui/app-installations/confirm-uninstall'
import { PagedResultInstallationModel_, InstallationModel } from '@reapit/foundations-ts-definitions'
import { handleUninstall, handleAfterClose } from '@/components/ui/app-installations/app-installations-modal'
import { useSelector, useDispatch } from 'react-redux'
import { selectInstallationAppData } from '@/selector/installations'
import { Modal } from '@reapit/elements'
import { generateColumns } from '@/components/ui/app-installations/installations'
import { AppDetailDataNotNull } from '@/reducers/client/app-detail'
import { DeveloperAppDetailState } from '@/reducers/developer'
import {
  ListingPreviewSection,
  AuthenticationSection,
  SummarySection,
  InstallationsTableSection,
  PermissionsSection,
} from '../common/ui-sections'

export type AppContentProps = {
  appDetailState: DeveloperAppDetailState
}

interface HandleUninstallSuccessParams {
  handleAfterClose: any
  setUninstallApp: React.Dispatch<React.SetStateAction<InstallationModel | undefined>>
  developerId: string
  appId: string
  dispatch: Dispatch
}

export const handleUninstallSuccess = ({
  handleAfterClose,
  setUninstallApp,
  developerId,
  appId,
  dispatch,
}: HandleUninstallSuccessParams) => () => {
  handleAfterClose({ setUninstallApp })
  dispatch(
    appInstallationsRequestData({
      appId: [appId],
      pageNumber: 1,
      pageSize: GET_ALL_PAGE_SIZE,
      isInstalled: true,
      developerId: [developerId],
    }),
  )
}

export const handleOpenAppPreview = (appId: string) => () => {
  const url = `developer/apps/${appId}/preview`
  window.open(url, '_blank')
}

export const CustomUninstallCell: React.FC<{ onClick: () => void }> = ({ onClick }) => (
  <a onClick={onClick}>Uninstall</a>
)

const AppContent: React.FC<AppContentProps> = ({ appDetailState }) => {
  const appDetailData = appDetailState.data as AppDetailDataNotNull
  const { summary = '', authFlow = '', externalId = '', id = '', name = '', scopes = [] } = appDetailData
  const installationsData = useSelector(selectInstallationAppData) as PagedResultInstallationModel_
  const { data = [] } = installationsData
  const dispatch = useDispatch()
  const developerId = useSelector(selectDeveloperId) || ''

  const [uninstallApp, setUninstallApp] = React.useState<InstallationModel>()
  const columns = generateColumns(handleUninstall(setUninstallApp), CustomUninstallCell)()
  const isVisibleUninstallModal = Boolean(uninstallApp)

  return (
    <FlexContainerBasic flexColumn>
      {isVisibleUninstallModal && (
        <Modal renderChildren visible={isVisibleUninstallModal}>
          <ConfirmUninstall
            isSetAppDetailStaleAfterUninstallSuccess={false}
            appName={name}
            installationDetail={uninstallApp}
            afterClose={handleAfterClose({ setUninstallApp })}
            onUninstallSuccess={handleUninstallSuccess({
              handleAfterClose,
              setUninstallApp,
              developerId,
              appId: id,
              dispatch,
            })}
          />
        </Modal>
      )}
      <SummarySection summary={summary} />
      <ListingPreviewSection onClick={handleOpenAppPreview(id)} />
      <AuthenticationSection id={id} authFlow={authFlow} externalId={externalId} />
      <PermissionsSection permissions={scopes} />
      <InstallationsTableSection columns={columns} data={data} />
    </FlexContainerBasic>
  )
}

export default AppContent
