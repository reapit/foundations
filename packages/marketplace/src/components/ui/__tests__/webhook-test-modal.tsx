import React from 'react'
import { mount, shallow } from 'enzyme'
import {
  generateTopicOptions,
  fetchWebhookData,
  handleSubmitForm,
  closeResultModal,
  WebhookTestResultModal,
  WebhookTestModalProps,
  WebhookTestModal,
  GenerateTopicOptions,
  WebhookTestModalFooter,
  WebhookTestModalBody,
} from '../webhook-test-modal'
import * as TestModal from '../webhook-test-modal'
import { ReduxState } from '@/types/core'
import * as ReactRedux from 'react-redux'
import { developerState } from '@/sagas/__stubs__/developer'
import configureStore from 'redux-mock-store'
import { TopicItem } from '@/reducers/webhook-edit-modal'
import { requestWebhookData } from '@/actions/webhook-edit-modal'
import { developerWebhookPing, developerSetWebhookPingStatus } from '@/actions/developer'

const mockState = {
  developer: developerState,
} as ReduxState

const props: WebhookTestModalProps = {
  visible: true,
  afterClose: jest.fn(),
  webhookId: 'webhookId',
  closeModal: jest.fn(),
}

describe('WebhookTestModalFooter', () => {
  it('WebhookTestModalFooter should match snapshot', () => {
    expect(shallow(<WebhookTestModalFooter closeModal={jest.fn} />)).toMatchSnapshot()
  })
})

describe('WebhookTestModalBody', () => {
  it('WebhookTestModalBody should match snapshot', () => {
    expect(shallow(<WebhookTestModalBody topicOptions={[]} />)).toMatchSnapshot()
  })
})

describe('WebhookTestModal', () => {
  let store
  beforeEach(() => {
    /* mocking store */
    const mockStore = configureStore()
    store = mockStore(mockState)
    /* mocking useDispatch on our mock store  */
    // spyDispatch = jest.spyOn(ReactRedux, 'useDispatch').mockImplementation(() => store.dispatch)
  })
  it('should WebhookTestModal match a snapshot', () => {
    spyOn(TestModal, 'generateTopicOptions').and.returnValue([])
    expect(
      mount(
        <ReactRedux.Provider store={store}>
          <WebhookTestModal {...props} />
        </ReactRedux.Provider>,
      ),
    ).toMatchSnapshot()
  })

  it('should WebhookTestResultModal match a snapshot', () => {
    expect(
      mount(
        <ReactRedux.Provider store={store}>
          <WebhookTestResultModal />
        </ReactRedux.Provider>,
      ),
    ).toMatchSnapshot()
  })

  describe('generateTopicOptions', () => {
    it('should run correctly', () => {
      const params: GenerateTopicOptions = {
        topics: [
          {
            id: 'a',
            name: 'A',
          } as TopicItem,
        ],
        subcriptionTopicIds: ['a'],
      }
      const result = generateTopicOptions(params)
      const expected = [
        {
          value: 'a',
          label: 'A',
          description: 'A',
        },
      ]
      expect(result).toEqual(expected)
    })
  })

  describe('fetchWebhookData', () => {
    it('should run correctly', () => {
      const dispatch = jest.fn()
      const webhookId = 'webhookId'
      const fn = fetchWebhookData(dispatch, webhookId)
      fn()
      expect(dispatch).toBeCalledWith(requestWebhookData(webhookId))
    })
  })
  describe('handleSubmitForm', () => {
    it('should run correctly', () => {
      const dispatch = jest.fn()
      const webhookId = 'webhookId'
      const topicId = 'topicId'

      const fn = handleSubmitForm({ dispatch, webhookId })
      fn({ topicId })
      const params = {
        id: webhookId,
        topicId: topicId,
      }
      expect(dispatch).toBeCalledWith(developerWebhookPing(params))
    })
  })
  describe('closeResultModal', () => {
    it('should run correctly', () => {
      const dispatch = jest.fn()
      const fn = closeResultModal(dispatch)
      fn()
      expect(dispatch).toBeCalledWith(developerSetWebhookPingStatus(null))
    })
  })
})
