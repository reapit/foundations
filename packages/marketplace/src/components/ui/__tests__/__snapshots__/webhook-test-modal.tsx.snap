// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebhookTestModal should WebhookTestModal match a snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Component
    afterClose={[MockFunction]}
    closeModal={[MockFunction]}
    visible={true}
    webhookId="webhookId"
  >
    <Component
      afterClose={[MockFunction]}
      renderChildren={true}
      visible={true}
    />
  </Component>
</Provider>
`;

exports[`WebhookTestModal should WebhookTestResultModal match a snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Component>
    <Component
      renderChildren={true}
      visible={false}
    />
  </Component>
</Provider>
`;

exports[`WebhookTestModalBody WebhookTestModalBody should match snapshot 1`] = `
<Component
  body={
    <React.Fragment>
      <Unknown>
        <p>
          To test your Webhook subscription, please select a ‘Subscription Topic’ below:
        </p>
      </Unknown>
      <Unknown
        id="topicId"
        labelText="Subscription Topics"
        name="topicId"
        options={Array []}
        required={true}
      />
    </React.Fragment>
  }
/>
`;

exports[`WebhookTestModalFooter WebhookTestModalFooter should match snapshot 1`] = `
<Component
  footerItems={
    <React.Fragment>
      <Unknown
        className="mr-2"
        fullWidth={true}
        onClick={[Function]}
        type="button"
        variant="danger"
      >
        CANCEL
      </Unknown>
      <Unknown
        fullWidth={true}
        type="submit"
        variant="primary"
      >
        TEST
      </Unknown>
    </React.Fragment>
  }
/>
`;
